### ISSUE TITLE
Null pointer derefernce detected by GCC

### ISSUE THREAD

**Issue by iv-m** [august-alt/gpui#16]
When building `gpui 0.2.17-alt8` for sisyphus_riscv64 I encountered the following error:

```
In file included from /usr/include/string.h:548,
                 from /usr/include/qt5/QtCore/qarraydata.h:44,
                 from /usr/include/qt5/QtCore/qbytearray.h:46,
                 from /usr/include/qt5/QtCore/qvariant.h:44,
                 from /usr/include/qt5/QtCore/qlocale.h:43,
                 from /usr/include/qt5/QtCore/QLocale:1,
                 from /usr/src/RPM/BUILD/gpui-0.2.17/src/ldap/ad_config.h:33,
                 from /usr/src/RPM/BUILD/gpui-0.2.17/src/ldap/ad_display.cpp:27:
In function 'void* memcpy(void*, const void*, size_t)',
    inlined from 'QString object_sid_display_value(const QByteArray&)' at /usr/src/RPM/BUILD/gpui-0.2.17/src/ldap/ad_display.cpp:158:11:
/usr/include/bits/string_fortified.h:29:33: error: argument 1 null where non-null expected [-Werror=nonnull]
   29 |   return __builtin___memcpy_chk (__dest, __src, __len,
      |          ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
   30 |                                  __glibc_objsize0 (__dest));
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/bits/string_fortified.h:29:33: note: in a call to built-in function 'void* __builtin_memcpy(void*, const void*, long unsigned int)'
cc1plus: all warnings being treated as errors
```

**Comment by iv-m** [comment]
At the first glance, it does not seem to be false-positive. `src/ldap/ad_display.cpp` around line 158 reads:

```c++
    std::unique_ptr<dom_sid> sid;
    memcpy(sid.get(), sid_bytes.data(), sizeof(dom_sid));
```

Default costructor of unique_ptr "Constructs a std::unique_ptr that owns nothing", so it's `get()` method will probably return null pointer. Passing null pointer to `memcpy` is UB, and a certain crash when size is not 0, which is the case.

Arguably a better way of doing the trickery that was probably intended here is allocating a `dom_sid` on the stack, and then copying the bytes in:

```c++
    dom_sid sid;
    memcpy((void*)&sid, sid_bytes.data(), sizeof(dom_sid));
```

It also triggers a few UBs, but, at least, it does not crash in an obvious way.

**Comment by august-alt** [comment]
This issue addressed in commit 1a60a3f. I also added compiler checks to prevent similar issues in the future.

**Comment by august-alt** [comment]
Version 0.2.17-alt9 resolves this issue.
